sdk-docs/tutorials/flask.md

```diff
@@ -0,0 +1,491 @@
+log verbosity levels and to start a live debug session:
+
+```bash
+# Set your logging verbosity level to `DEBUG`:
+ubuntu@charm-dev-vm:~$  juju model-config logging-config="<root>=WARNING;unit=DEBUG"
+
+# Start a live debug session:
+ubuntu@charm-dev-vm:~$  juju debug-log
+```
+
+In your old VM shell, use Juju to deploy your charm. If all has gone well, you
+should see your App and be able to connect to the sample Flask web server:
+
+```bash
+# Deploy the PaaS app charmer Flask charm for the sample-flask Flask application
+# We need to supply the resources to Juju, including the sample-flask rock we just created and pushed to the microk8s registry
+# and the image prom/statsd-exporter:v0.24.0 used for adding observability to the Flask application
+ubuntu@charm-dev-vm:~/sample-flask$ juju deploy ./charm/sample-flask_ubuntu-22.04-amd64.charm sample-flask --resource flask-app-image=localhost:32000/sample-flask:main
+Located local charm "sample-flask", revision 0
+Deploying "sample-flask" from local charm "sample-flask", revision 0 on ubuntu@22.04/stable
+
+# Check the deployment status
+# (use --watch 1s to update it automatically at 1s intervals):
+ubuntu@charm-dev-vm:~/sample-flask$ juju status
+Model        Controller  Cloud/Region        Version  SLA          Timestamp
+welcome-k8s  microk8s    microk8s/localhost  3.1.7    unsupported  10:14:28Z
+
+App           Version  Status  Scale  Charm         Channel  Rev  Address         Exposed  Message
+sample-flask           active      1  sample-flask             0  10.152.183.175  no
+
+Unit             Workload  Agent  Address      Ports  Message
+sample-flask/0*  active    idle   10.1.30.206
+```
+
+Finally, test that the service works by using curl to send a request to the
+Flask service:
+```bash
+# Unit IP (10.1.30.206) can be learned from the juju status output
+ubuntu@charm-dev-vm:~/sample-flask$ curl 10.1.30.206:8000
+<h1>Hello</h1>
+```
+
+## Enable `juju config sample-flask language=<language>`
+
+Let's now expand the functionality of the `sample-flask` Flask application to
+return the greeting in different languages. To achieve this, we will introduce a
+new configuration for changing the language selection. The PaaS app charmer
+Flask framework offers a variety of standard configurations; these can be
+explored by executing the `juju config sample-flask` command. And it's very
+simple to integrating custom configurations, such as the required language
+option mentioned above.
+
+Let's first checkout the `feat-translation` branch of the `sample-flask`
+repository to load the new version of the `sample-flask` application.
+
+```bash
+ubuntu@charm-dev-vm:~/sample-flask$ git checkout feat-translation
+Branch 'feat-translation' set up to track remote branch 'feat-translation' from 'origin'.
+Switched to a new branch 'feat-translation'
+```
+
+Since the `sample-flask` Flask application has changed, we need to repack the
+rock image and push the new image to the registry.
```

Comment:
The fact that here we are repacking the rock argues against treating the rock step as mere setup. What's interesting though is that we're suddenly making rockcraft part of the charm SDK. It's a new thing, but makes sense, so why not... (but it _is_ in fact a new thing, so we have to make sure it has the correct reflexes in our reference docs or somewhere).

Reply:
As David explained, this makes sense -- since when you're using an extension so much of the rocking and charming work has been done for you, you can focus more on developing your charm's workload further, and that also raises the question of what to do when you've updated it.